README.md

```diff
@@ -243,8 +243,47 @@ evaluation.run(model)
 </details>
 
 <details>
-  <summary>  Saving retrieval task predictions </summary>
+  <summary>  Using a cross encoder for reranking</summary>
+
+
+### Using a cross encoder for reranking
+
+To use a cross encoder for reranking, you can directly use a CrossEncoder from SentenceTransformers. The following code shows a two-stage run with the second stage reading results saved from the first stage. 
+
+```python
+from mteb import MTEB
+import mteb
+from sentence_transformers import CrossEncoder, SentenceTransformer
+
+cross_encoder = CrossEncoder("cross-encoder/ms-marco-TinyBERT-L-2-v2")
+dual_encoder = SentenceTransformer("all-MiniLM-L6-v2")
+
+tasks = mteb.get_tasks(tasks=["NFCorpus"], languages=["eng"])
+
+subset = "default" # subset name used in the NFCorpus dataset
+eval_splits = ["test"]
+
+evaluation = MTEB(tasks=tasks)
+evaluation.run(
+    dual_encoder,
+    eval_splits=eval_splits,
+    save_predictions=True,
+    output_folder="results/stage1",
+)
+evaluation.run(
```

Comment:
hmm the problem with this as it is currently is that it will produce a results file (with the model_meta.json) as if "cross-encoder/ms-marco-TinyBERT-L-2-v2" got the score without acknowledging the effect of "all-MiniLM-L6-v2". It is totally fine for now to ignore that, but we might end up with it being an issue.

An option might be to grab the model_meta.json from `results/stage1` and transfer the information (in some way) to the results/stage2. Not sure what the ideal result should be essentially want to save 1) the results as usual, 2) the model meta for the cross-encoder, and 3) the model meta for the dual encoder. 3) could of course in some way be included in 2) (but the logic is located in quite different places).

Reply:
Yeah agree this would be something that would be nice to have but not trivial to grab.